libs/platform/src/lib/components/form/time-picker/time-picker.component.ts

```diff
@@ -0,0 +1,185 @@
+    @Input()
+    displayMinutes = true;
+
+    /** @Input When set to false, hides the input for hours. */
+    @Input()
+    displayHours = true;
+
+    /** @Input Default time picker placeholder which is set dependant on the hours, minutes and seconds.
+     * Otherwise It can be set to a default value
+     */
+    @Input()
+    placeholder: string = this.getPlaceholder();
+
+    /** Aria label for the time picker input. */
+    @Input()
+    timePickerInputLabel = 'Time picker input';
+
+    /** Whether a null input is considered valid(success). */
+    @Input()
+    allowNull = true;
+
+    /** Defines if time component should be used with tablet mode */
+    @Input()
+    tablet = false;
+
+    /**
+     *  The placement of the popover. It can be one of: top, top-start, top-end, bottom,
+     *  bottom-start, bottom-end, right, right-start, right-end, left, left-start, left-end.
+     */
+    @Input()
+    placement: Placement = 'bottom-start';
+
+    @Input()
+    set state(state: Status) {
+        this._state = state ? state : 'default';
+    }
+
+    get state(): Status {
+        if (this.timePickerComponent && this.timePickerComponent.isInvalidTimeInput) {
+            // if any other error from core timePicker
+            return 'error';
+        }
+        if (this.status) {
+            return this.status;
+        }
+        return this._state;
+    }
+
+    /**
+     * Whether AddOn Button should be focusable, set to true by default
+     */
+    @Input()
+    buttonFocusable = true;
+
+    /**
+     * @Input when set to true, time inputs won't allow to have 1 digit
+     * for example 9 will become 09
+     * but 12 will be kept as 12.
+     */
+    @Input()
+    keepTwoDigitsTime = false;
+
+    /** Event emitted when the state of the isOpen property changes. */
+    @Output()
+    readonly isOpenChange = new EventEmitter<boolean>();
+
+    @ViewChild(TimePickerComponent)
+    timePickerComponent: TimePickerComponent;
+
+    /**
+     * @hidden
+     * The state of the form control - applies css classes.
+     * Can be `success`, `error`, `warning`, `information` or blank for default.
+     */
+    private _state: Status;
+
+    /**
+     * Method that handles changes when popover is opened or closed.
+     */
+    handleOpenChange(open: boolean): void {
+        this.isOpenChange.emit(open);
+    };
+
+    /**
+     * logic to handle validation from both platform forms and core datetiimepicker
+     * @param value inputted
+     */
+    handleTimeChange(value: TimeObject): void {
+        if (this.timePickerComponent) {
+            if (this.timePickerComponent.isInvalidTimeInput) {
+                this.state = 'error';
+            } else {
+                if (!this.timePickerComponent.time && !this.allowNull) {
+                    this.state = 'error'; // null value in not allowNull should throw error
+                } else {
+                    this.state = undefined; // resetting to default state
+                }
```

Comment:
If you want you can change this to ternary expression to make it more clear:
```
!this.timePickerComponent.time && !this.allowNull ? this.state = 'error' : this.state = undefined
```

Reply:
@InnaAtanasova for me it's more readable that way :) but not a big deal, I can change it if you think it will be better.

Reply:
my message starts with `If you want...`. I never insist on other people writing code the way I do :)

Reply:
And I didn't request changes, meaning all my comments are "debatable". If I think something must be fix it will be evident :P (just for future reference)

Reply:
I didn't approve because there was no approval from Platform team and our agreement is to not approve Platform PRs without at least 1 approval from your team.